src/ServiceControl/Operations/ErrorIngestionCustomCheck.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ServiceControl.Operations
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.CustomChecks;
+
+    class ErrorIngestionCustomCheck : CustomCheck
+    {
+        readonly State criticalErrorHolder;
+
+        public ErrorIngestionCustomCheck(State criticalErrorHolder) 
+            : base("Failed message ingestion process", "ServiceControl", TimeSpan.FromSeconds(5))
+        {
+            this.criticalErrorHolder = criticalErrorHolder;
+        }
+
+        public override Task<CheckResult> PerformCheck()
+        {
+            var failure = criticalErrorHolder.GetLastFailure();
+            if (failure == null)
+            {
+                return Task.FromResult(CheckResult.Pass);
+            }
+
+            return Task.FromResult(CheckResult.Failed(failure));
```

Comment:
If failure is cached this task can be as well

Reply:
I think as we expect this state to be not that common I think we should be fine with only caching the success result (for readability of the code).